src/frontend/src/binder/expr/function/builtin_scalar.rs

```diff
@@ -434,6 +434,57 @@ impl Binder {
                 ("inner_product", raw_call(ExprType::InnerProduct)),
                 ("vector_norm", raw_call(ExprType::L2Norm)),
                 ("l2_normalize", raw_call(ExprType::L2Normalize)),
+                ("subvector", guard_by_len(|_, [vector_expr, start_expr, len_expr]| {
+                    let dimensions = if let DataType::Vector(length) = vector_expr.return_type() {
+                        length as i32
+                    } else {
+                        return Err(ErrorCode::BindError("subvector expects `vector(dim)` input".into()).into());
+                    };
+                    let start = start_expr
+                        .try_fold_const()
+                        .transpose()?
+                        .and_then(|datum| match datum {
+                            Some(ScalarImpl::Int32(v)) => Some(v),
+                            _ => None,
+                        })
+                        .ok_or_else(|| ErrorCode::ExprError("`start` must be an Int32 constant".into()))?;
+
+                    let len = len_expr
+                        .try_fold_const()
+                        .transpose()?
+                        .and_then(|datum| match datum {
+                            Some(ScalarImpl::Int32(v)) => Some(v),
+                            _ => None,
+                        })
+                        .ok_or_else(|| ErrorCode::ExprError("`count` must be an Int32 constant".into()))?;
+                    if len < 1 {
+                        return Err(ErrorCode::InvalidParameterValue(format!("vector must have at least 1 dimension, not {}", len)).into());
+                    }
+
+                    let end = start + len - 1;
+
+                    if start < 1 || end < 1 || start > dimensions || end > dimensions {
+                        return Err(ErrorCode::InvalidParameterValue(format!(
+                                "vector slice range out of bounds: start={}, end={}, valid range is [1, {}]",
+                                start,
+                                end,
+                                dimensions
+                            )).into());
+                    }
+
+                    let dim = (end - start + 1) as usize;
```

Comment:
The check on `dim` seems unnecessary, because dim is strictly equal to `len` in current code.